### PR TITLE
Update custom store readme to use thunks instead of controls

### PR DESCRIPTION
**Pull request by SantosGuillamot**
## What?
Update the custom store explanation to use [thunks](https://github.com/WordPress/gutenberg/blob/trunk/docs/how-to-guides/thunks.md#thunks-api) instead of controls.

## Why?
If I am not mistaken, now that thunks are supported, it should be the recommended way of handling these use cases as explained [here](https://github.com/WordPress/gutenberg/blob/trunk/docs/how-to-guides/thunks.md#thunks-api).

## How?
I just updated the code examples in the readme.

### CODE REVIEW

File: packages/data/README.md [hunk 1]
```diff
@@ -133,8 +120,12 @@ A **resolver** is a side-effect for a selector. If your selector result may need
...
+
 #### `controls`
 
+> To handle asynchronous data flows, it is recommended to use [thunks](https://github.com/WordPress/gutenberg/blob/trunk/docs/how-to-guides/thunks.md#thunks-can-be-async) instead of `controls` now that they are supported.
+
 A **control** defines the execution flow behavior associated with a specific action type. This can be particularly useful in implementing asynchronous data flows for your store. By defining your action creator or resolvers as a generator which yields specific controlled action types, the execution will proceed as defined by the control handler.
 
 The `controls` option should be passed as an object where each key is the name of the action type to act upon, the value a function which receives the original action object. It should returns either a promise which is to resolve when evaluation of the action should continue, or a value. The value or resolved promise value is assigned on the return value of the yield assignment. If the control handler returns undefined, the execution is not continued.
```
Comment (SantosGuillamot) [inline packages/data/README.md]: I am not sure what to do with the `controls` section. Should it be refactored? Removed? I am not familiar with how controls are used.

File: packages/data/README.md [hunk 2]
```diff
@@ -262,7 +253,7 @@ The data module shares many of the same [core principles](https://redux.js.org/i
...
 
-The data module also has built-in solutions for handling asynchronous side-effects, through [resolvers](#resolvers) and [controls](#controls). These differ slightly from [standard redux async solutions](https://redux.js.org/advanced/async-actions) like [`redux-thunk`](https://github.com/gaearon/redux-thunk) or [`redux-saga`](https://redux-saga.js.org/).
+The data module also has built-in solutions for handling asynchronous side-effects, through [resolvers](#resolvers) and [thunks](https://github.com/WordPress/gutenberg/blob/trunk/docs/how-to-guides/thunks.md#thunks-can-be-async). These differ slightly from [standard redux async solutions](https://redux.js.org/advanced/async-actions) like [`redux-thunk`](https://github.com/gaearon/redux-thunk) or [`redux-saga`](https://redux-saga.js.org/).
```
Comment (SantosGuillamot) [inline packages/data/README.md:260]: Should we update this sentence or does it still apply?

> These differ slightly from [standard redux async solutions](https://redux.js.org/advanced/async-actions) like [`redux-thunk`](https://github.com/gaearon/redux-thunk) or [`redux-saga`](https://redux-saga.js.org/).

File: packages/data/README.md (hunk 1 above)
Comment (youknowriad) [reply packages/data/README.md]: Can we hide it in a "details" kind of? and mark it as "deprecated"?

Comment (SantosGuillamot) [reply packages/data/README.md]: I've tried that in this commit: https://github.com/WordPress/gutenberg/pull/67006/commits/6f24f40701503a5e2e16dd5f017130a841139de4?short_path=f874ff4#diff-f874ff4a9c60876a3d01ac27950807424eead31c2cd914daa34136108bd03e0f. Let me know if that's not what you had in mind. I am not sure if it needs to be changed somewhere else.

Comment (youknowriad) [reply packages/data/README.md]: Yes, that's basically what I had in mind. I wonder if will translate properly to the block editor handbook when published.

Comment (SantosGuillamot) [reply packages/data/README.md]: I saw that the Interactivity API readmes have a bunch of `<details>` ([example](https://github.com/WordPress/gutenberg/blob/6f24f40701503a5e2e16dd5f017130a841139de4/docs/reference-guides/interactivity-api/api-reference.md#wp-context)) and they seem to look fine in the block editor handbook ([example](https://developer.wordpress.org/block-editor/reference-guides/interactivity-api/api-reference/#wp-context)). Although it's true in this case I'm adding different elements.

**Review by youknowriad — APPROVED** [review]
I didn't test the example but it looks good at first sight.

File: packages/data/README.md (hunk 1 above)
Comment (gziolo) [reply packages/data/README.md]: It makes perfect sense to mark the controls as deprecated 👍🏻

File: packages/data/README.md (hunk 2 above)
Comment (gziolo) [reply packages/data/README.md:260]: @jsnajdr might know the answer. I don't think it's a blocker for merging, though.

**Review by gziolo — APPROVED** [review]
> I didn't test the example but it looks good at first sight.

Yes, it's difficult to test as the endpoint doesn't exist. There are some unit tests with examples like the following that I used for comparison:

https://github.com/WordPress/gutenberg/blob/a750125b0210b9d906efcec2d807477590e4412f/packages/data/src/components/use-select/test/suspense.js#L165-L192

Overall, it's a good improvement that promotes thunks.

**Comment by Mamaduka** [comment]
> Yes, it's difficult to test as the endpoint doesn't exist.

I think that's always been a barrier for consumers who wanted to get started with data stores. The primary example should be copy-pastable into plugin files to get you started.

We could follow up on this separately. Maybe @ryanwelcher has some ideas :)

**Pull request merged**